Fixed output frequency and reference time.

feng045 opened the pull request:
1. Updated process_zarr.py to:
- Fixed bug where output frequency does not match input.
- Added --ref_time input argument for output netCDF time encoding. Default (not supplied) is ‘2000-01-01T00:00:00’. The time encoding is now ‘seconds since {ref_time}’.

2. Updated run_process_zarr.sh to show example with --ref_time.

feng045 committed:
Fixed output frequency and reference time.

1. Updated process_zarr.py to:
- Fixed bug where output frequency does not match input.
- Added --ref_time input argument for output netCDF time encoding. Default (not supplied) is ‘2000-01-01T00:00:00’. The time encoding is now ‘seconds since {ref_time}’.

2. Updated run_process_zarr.sh to show example with --ref_time.

feng045 committed:
Corrected output attributes.

laurapaccini commented:
Thanks @feng045 for the edits. I tested it  and noticed that when defining the initial 'time_steps' based on the file time ('time0=time_in[0]') it doesn't always match the expected frequency. For example, if I'd like 3h data, I'd expect outputs at 0,3,6,9,12,..hrs. However, if the input file starts at '2000-1-10 05:00', it gives me different hour outputs. I'll correct this just by using the reference_time as time0.
Also, I still receive this message sometimes (~10 times) when running the script in parallel: 
"2024-10-30 15:20:09,844 - distributed.worker - ERROR - Failed to communicate with scheduler during heartbeat..."
This does not seem to affect the files, just delays the process. The output looks good.

Commit:
Fix initial time_steps calculation

- Adjust initial time_steps to use reference_time as time0 to ensure expected frequency output.
- Note: occasional communication failures with the scheduler during parallel execution do not affect file outputs.

laurapaccini reviewed (changes requested):
I updated the time step handling to use reference_time as time0 for consistency. However, I’m seeing occasional communication conflicts in parallel processing. I'd suggest we further investigate and address these to ensure robustness.

feng045 commented:
Thanks for catching that @laurapaccini.  In that situation, it's more because of the weird "non-standard" first time of the existing model output.  But I think your solution would mitigate that. 

As for the Dask warning, I see that quite often but don't know exactly why or how to configure Dask to avoid those warnings. But since it does not affect the outputs, I usually just ignore them.
